lib/UserBackend.php

```diff
@@ -248,6 +256,35 @@ public function getDisplayNames($search = '', $limit = null, $offset = null): ar
 		}
 	}
 
+	/**
+	 * Get a list of all users with their email and display name
+	 *
+	 * @return array<string, array{email: string, displayname: string}>
+	 */
+	public function getAllGuestAccounts(?int $limit = null, ?int $offset = null): array {
+		if (!$this->allowListing) {
+			return [];
+		}
+
+		$query = $this->dbConn->getQueryBuilder();
+		$query->select('uid', 'email', 'displayname')
+			->from('guests_users')
+			->orderBy('uid_lower', 'ASC')
+			->setMaxResults($limit)
+			->setFirstResult($offset);
+
+		$result = $query->execute();
+		$users = [];
+		while ($row = $result->fetch()) {
+			$users[(string)$row['uid']] = [
+				'email' => (string)$row['email'],
+				'displayname' => (string)$row['displayname'],
+			];
```

Comment:
nit: a generator may make sense here: Do all guests fit in memory?

Reply:
getDisplayNames() above is basically doing the same thing. Just had to do an array as value so we can have email + displayname